lib/SecretsCrypto/sf-common.c

```diff
@@ -10,6 +10,15 @@ GVariant *_sf_variant_new_bytes_or_empty(GBytes *bytes)
             sizeof(guchar));
 }
 
+GBytes *_sf_bytes_new_from_variant(GVariant *variant)
+{
+    gconstpointer data;
+    gsize n_elements;
+
+    data = g_variant_get_fixed_array(variant, &n_elements, sizeof(guchar));
+    return g_bytes_new(data, n_elements * sizeof(guchar));
```

Comment:
Something like this would avoid copying the data:
```c
return g_bytes_new_with_free_func(data, n_elements,
    (GDestroyNotify) g_variant_unref, g_variant_ref(variant));
```